src/struc/model.jl

```diff
@@ -0,0 +1,826 @@
+export StrucElement,
+    AbstractAtom,
+    Atom,
+    DisorderedAtom,
+    AbstractResidue,
+    Residue,
+    DisorderedResidue,
+    Chain,
+    Model,
+    Structure,
+    AtomList,
+    ResidueList,
+    ChainList,
+    ModelList,
+    StrucElementOrList,
+    ishetatom,
+    getserial,
+    getatomname,
+    getaltlocid,
+    getresname,
+    getchainid,
+    getresnumber,
+    getinscode,
+    getx,
+    gety,
+    getz,
+    getcoords,
+    getoccupancy,
+    gettempfac,
+    getelement,
+    getcharge,
+    setx!,
+    sety!,
+    setz!,
+    setcoords!,
+    getdefaultaltlocid,
+    getdefaultatom,
+    getaltlocids,
+    ishetero,
+    setdefaultaltlocid!,
+    getresid,
+    ishetres,
+    getatomnames,
+    getdisorderedres,
+    getdefaultresname,
+    getdefaultresidue,
+    getresnames,
+    setdefaultresname!,
+    getresids,
+    getmodelnumber,
+    getchainids,
+    getstrucname,
+    getmodelnumbers,
+    setstrucname!,
+    applyselectors,
+    applyselectors!,
+    collectresidues,
+    collectatoms,
+    countmodels,
+    countchains,
+    countresidues,
+    countatoms,
+    isdisordered,
+    organise,
+    formatomlist,
+    choosedefaultaltlocid,
+    organisemodel,
+    organisestruc,
+    stdatomselector,
+    hetatomselector,
+    atomnameselector,
+    calpha_atom_names,
+    calphaselector,
+    backbone_atom_names,
+    backboneselector,
+    heavy_atom_names,
+    heavyatomselector,
+    resnameselector,
+    water_res_names,
+    waterselector,
+    stdresselector,
+    hetresselector,
+    disorderselector,
+    hydrogenselector
+
+
+import Base: getindex,
+    setindex!,
+    length,
+    start,
+    next,
+    done,
+    eltype,
+    show
+
+
+"""A protein structural element."""
+abstract StrucElement
+
+
+"""An atom represented in a PDB file - either an `Atom` or a
+`DisorderedAtom`."""
+abstract AbstractAtom <: StrucElement
+
+"""An atom record from a PDB file."""
+immutable Atom <: AbstractAtom
+    het_atom::Bool
+    serial::Int
+    name::AbstractString
+    alt_loc_id::Char
+    res_name::AbstractString
+    chain_id::Char
+    res_no::Int
+    ins_code::Char
+    coords::Array{Float64,1}
```

Comment:
Cool. Am I right in thinking that making it an `NTuple` would prevent changing the coordinates without re-instantiating, as the type is immutable?
With the coords as an `Array` I was conveniently just changing the elements for `setx!` etc., though this could be hacky and un-Julian, I don't know.
I guess it depends whether we want people to be able to change the atomic coordinates quickly - I suppose if you're doing something like MD simulation you might have a new type, something like:
```
type MDAtom
    atom::Atom
    current_coords::NTuple{3,Float64}
end
```
In this case the `coords` field in the `Atom` type would represent the (invariant) crystal coordinates, and so could happily remain immutable.